odl/solvers/advanced/forward_backward.py

```diff
@@ -0,0 +1,173 @@
+# Copyright 2014-2016 The ODL development group
+#
+# This file is part of ODL.
+#
+# ODL is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# ODL is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with ODL.  If not, see <http://www.gnu.org/licenses/>.
+
+"""Optimization methods based on a forward-backward splitting scheme."""
+
+
+# Imports for common Python 2/3 codebase
+from __future__ import print_function, division, absolute_import
+from future import standard_library
+standard_library.install_aliases()
+
+from odl.operator import Operator
+
+
+__all__ = ('forward_backward_pd',)
+
+
+def forward_backward_pd(x, prox_f, prox_cc_g, L, grad_h, tau, sigma, niter,
+                        callback=None, **kwargs):
+    """The forward-backward primal-dual splitting algorithm.
+
+    The algorithm minimizes the sum of several convex functionals composed with
+    linear operators,
+
+        ``min_x f(x) + sum_i g_i(L_i x) + h(x)``,
```

Comment:
I think the order is a bit arbitrary as well... we might as well also take the order
```
min_x h(x) + f(x) + sum_i g_i(L_i x)
```
because, in some sense, the functionals are then in the order of "niceness" (h is differentiable, so nicest while g is composed with the linear operator and also involved in the infimal convolution if `l_i` is set).

I think we should stick to the same conventions as the articles; at least it makes comparing the algorithm with the article more easy.